src/app/shared-links/LinksTable.tsx

```diff
@@ -0,0 +1,151 @@
+'use client';
+import {
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TablePagination,
+  TableRow,
+} from '@mui/material';
+import React from 'react';
+
+import { SHLinkMiniDto } from '@/domain/dtos/shlink';
+
+interface Column {
+  id: keyof SHLinkMiniDto;
+  label: string;
+  minWidth?: number;
+  align?: 'right' | 'left' | 'center';
+  // string | boolean | Date | SHLinkFileDto[]
+  format?: (value: SHLinkMiniDto[keyof SHLinkMiniDto]) => string;
+}
+
+const columns: readonly Column[] = [
+  { id: 'id', label: 'Id', minWidth: 100 },
+  { id: 'name', label: 'Name', minWidth: 100 },
+  { id: 'url', label: 'Url', minWidth: 100 },
+  { id: 'managementToken', label: 'Management Token', minWidth: 100 },
+  {
+    id: 'expiryDate',
+    label: 'Expiry Date',
+    minWidth: 80,
+    format: (value: Date) => value.toLocaleDateString(),
+  },
+  {
+    id: 'passwordRequired',
+    label: 'Password',
+    minWidth: 80,
+    // format: (value: boolean) => (!value?`Not`:`` value),
+    format: (value: boolean) => `${!value ? 'Not' : ''} Required`,
+  },
+];
+
+function createData(data: SHLinkMiniDto): SHLinkMiniDto {
+  return { ...data };
+}
+
+const apiDataMock: SHLinkMiniDto[] = [
```

Comment:
since the API is available would it not be better to connect to it already?

Reply:
Yes, working on that.

Reply:
Yes, working on that.